08_cell_line_prediction/nbconverted/download_data.py

```diff
@@ -0,0 +1,353 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Download and preprocess CCLE data
+
+# In[1]:
+
+
+import os
+import sys
+
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+import pancancer_evaluation.config as cfg
+
+
+# ### Download cell line info and expression data
+
+# In[2]:
+
+
+if os.path.isfile(cfg.ccle_sample_info):
+    ccle_sample_info_df = pd.read_csv(cfg.ccle_sample_info, sep=',', index_col=0)
+else:
+    print('Loading sample info from CCLE download page...', file=sys.stderr)
+    # URL from CCLE public download data:
+    # https://depmap.org/portal/download/
+    ccle_sample_info_df = pd.read_csv(
+        'https://ndownloader.figshare.com/files/35020903',
```

Comment:
Figshare data resources are [intended to be archival](https://help.figshare.com/article/preservation-and-continuity-of-access-policy), and I've seen links like these ones in papers, so I think it's unlikely that the link will change or break on their end. It is possible for the uploader (the Broad in this case) to [delete a Figshare dataset](https://help.figshare.com/article/can-i-edit-or-delete-my-research-after-it-has-been-made-public) but I would imagine that's unlikely for this dataset since a lot of people use it.

I'll include the main CCLE data download URL in a comment, just in case.